src/pages/Map.js

```diff
@@ -22,14 +24,28 @@ export default function Map() {
     });
   };
 
+  //TODO: added just in case we want to access the googleMapApiInternals
+  const handleApiLoaded = (map, maps) => {
+    // use map and maps objects
+  };
+
+  /*When the map moves it resets the center to recenter 
+  we are updating the state with the setRecenter which updates context and updates locationData */
+  const handleChange = (e) => {
+    setRecenter({ lat: e.center.lat, lng: e.center.lng });
+  };
+
   return (
     <div className="map">
       <GoogleMapReact
         bootstrapURLKeys={{ key: API_KEY }}
+        center={recenter}
         defaultCenter={coordinates}
         defaultZoom={zoomLevel}
         yesIWantToUseGoogleMapApiInternals
         onChildClick={onChildClick}
+        onChange={(e) => handleChange(e)}
```

Comment:
Consider using `onDragEnd` instead of `onChange` ... `onDragEnd` will only fire once whereas `onChange` may fire multiple times as the user is dragging the map. Docs here: https://github.com/google-map-react/google-map-react/blob/HEAD/API.md#ondragend-map--void

Reply:
great!